docs/transform/routing_transformer.ipynb

```diff
@@ -0,0 +1,457 @@
+    "2. `tag_inserted_swaps: bool`: whether or not a `cirq.RoutingSwapTag` should be attached to inserted swap operations in order to distinguish inserted SWAP gates by the routing procedure from SWAP gates part of the input circuit.\n",
+    "3. `initial_mapper: Optional['cirq.AbstractInitialMapper']`: an initial mapping strategy (placement) of logical qubits in the circuit onto physical qubits on the device. If not provided, defaults to an instance of `cirq.LineInitialMapper`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "I6csX1EJmAEU"
+   },
+   "source": [
+    "Here is an example of routing the same circuit on the same device with non-default optional arguments."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "DU8lHNiK4eCL"
+   },
+   "outputs": [],
+   "source": [
+    "# Use a hard-coded initial mapping strategy of logical to physical that places q0, q1, q2, q3 onto\n",
+    "# Grid(3, 5), Grid(3, 6), Grid(4, 5), Grid(4, 6), respectively\n",
+    "gq = cirq.GridQubit(3, 5)\n",
+    "hc_initial_mapper = cirq.HardCodedInitialMapper(\n",
+    "    {q[0]: gq, q[1]: gq + (0, -1), q[2]: gq + (-1, 0), q[3]: gq + (-1, -1)}\n",
+    ")\n",
+    "print(hc_initial_mapper)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "RMGa1h2uoKui"
+   },
+   "outputs": [],
+   "source": [
+    "routed_circuit = router(circuit, lookahead_radius = 5, tag_inserted_swaps=True, initial_mapper=hc_initial_mapper)\n",
+    "print(routed_circuit)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "bbfue03H_2Lc"
+   },
+   "source": [
+    "### Unitary Equivalence"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "l1RLuUE02-Qd"
+   },
+   "source": [
+    "It is often the case that the routing process will return a routed circuit that is unitarily equivalent to the input circuit but with the order of the qubits permuted. This is handled by calling the method of the router `route_circuit` with the input ciruit as an argument instead of calling the router. \n",
+    "\n",
+    "In addition to returning the routed circuit, `route_circuit` also returns the initial mapping of logical to physical qubits and the permutation of qubits caused by insertion of SWAP gates. For example, one way to recover the input circuit's unitary is to permute the routed circuit's qubits as in the example below:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "R9ikw2Jz3HND"
+   },
+   "outputs": [],
+   "source": [
+    "# Example with appending SwapPermutation gate at end and permuting initial order of qubits.\n",
```

Comment:
Do you mean a `QubitPermutationGate` instead of `SwapPermutationGate` ?
```suggestion
    "# Example with appending QubitPermutationGate gate at end and permuting initial order of qubits.\n",
```